### PR TITLE
Add support for Windows E2E testing

### PR DESCRIPTION
**Pull request by oprinmarius**
- Add packer template for Windows
- Add Windows to build_VHD's Jenkins job
- Add oe-engine template with custom VHD for Windows
- Add Windows nodes to custom_agent Jenkins job

### CODE REVIEW

File: .jenkins/provision/templates/packer/packer-win.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      ],
+      "command": "ansible-playbook -e \"ansible_user=packer ansible_password={{.WinRMPassword}} ansible_become_pass={{.WinRMPassword}} ansible_connection=winrm ansible_winrm_server_cert_validation=ignore ansible_winrm_transport=ntlm\" {{ user `playbook_file` }}"
+    },
+    {
+      "type": "powershell",
+      "inline": [
+          "  while ((Get-Service RdAgent).Status -ne 'Running') { Start-Sleep -s 5 }",
+          "  while ((Get-Service WindowsAzureTelemetryService).Status -ne 'Running') { Start-Sleep -s 5 }",
+          "  while ((Get-Service WindowsAzureGuestAgent).Status -ne 'Running') { Start-Sleep -s 5 }",
+          "& $env:SystemRoot\\System32\\Sysprep\\Sysprep.exe /oobe /generalize /quiet /quit",
```
Comment (johnkord) [inline .jenkins/provision/templates/packer/packer-win.json]: Fancy work here!

File: .jenkins/provision/templates/packer/packer-win.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    "winrm_username": "Administrator",
+
+    "location": "{{user `location`}}",
+    "vm_size": "{{user `vm_size`}}"
+  }],
+  "provisioners": [
+    {
+        "type": "powershell",
+        "inline": [
+          "(Invoke-WebRequest -Headers @{\"Metadata\"=\"true\"} -UseBasicParsing \"http://169.254.169.254/metadata/instance/network/interface/0/ipv4/ipAddress/0/publicIpAddress?api-version=2017-08-01&format=text\").Content | Out-File -Append -Encoding utf8 C:/Windows/Temp/ip-address"
```
[the remark this answers is not in the document]
Comment (oprinmarius) [reply .jenkins/provision/templates/packer/packer-win.json]: This is the IP address of the VM that packer creates.
Unfortunately packer does not expose via variables the IP address of the VM that it creates.  We have to retrieve it either from packer log from /tmp folder inside docker container , or as we do it here :
  - We use a powershell provider to connect to instance and retrieve it's IP address from Azure metadata

[the remark this answers is not in the document]
Comment (oprinmarius) [reply .jenkins/provision/templates/packer/packer-win.json]: We need to get this IP address to generate the inventory for ansible , so we can run our playbook on the newly created packer instance

File: scripts/ansible/oe-windows-acc-setup.yml [hunk 3]
```diff
@@ -36,8 +36,8 @@
     - name: OE setup | Run the install-windows-prereqs.ps1 script (this may take a while)
       script: ../install-windows-prereqs.ps1
         -InstallPath         "C:\openenclave"
-        -LaunchConfiguration "{{ SGX1FLC if dcap_testing_node is defined and dcap_testing_node == true else SGX1 }}"
-        -DCAPClientType      "{{ Azure }}"
+        -LaunchConfiguration "{{ 'SGX1FLC' if dcap_testing_node is defined and dcap_testing_node == true else 'SGX1' }}"
```
Comment (johnkord) [inline scripts/ansible/oe-windows-acc-setup.yml]: Thanks for updating this with the new changes that went in yesterday!

**Comment by oprinmarius** [comment]
@johnkord 
Here is a full run of E2E testing
https://oe-jenkins.eastus.cloudapp.azure.com/blue/organizations/jenkins/OpenEnclave-E2E-Testing/detail/OpenEnclave-E2E-Testing/49/pipeline

**Comment by oprinmarius** [comment]
bors try

**Pull request merged**